subsys/net/lib/sockets/sockets_tls.c

```diff
@@ -440,7 +440,7 @@ static struct tls_context *tls_clone(struct tls_context *source_tls)
 	memcpy(&target_tls->options, &source_tls->options,
 	       sizeof(target_tls->options));
 
-#if defined(MBEDTLS_X509_CRT_PARSE_C)
+#if defined(MBEDTLS_X509_CRT_PARSE_C) || defined(MBEDTLS_SSL_SERVER_NAME_INDICATION)
```

Comment:
Will this actually work? `mbedtls_ssl_set_hostname()` seems to be compiled out if MBEDTLS_X509_CRT_PARSE_C is not enabled (https://github.com/zephyrproject-rtos/mbedtls/blob/8df2f8e7b9c7bb9390ac74bb7bace27edca81a2b/library/ssl_tls.c#L3909)

Reply:
It will work if you consider the change this PR makes also to `west.yml` that points to a pull request against `mbedtls` to that specific section of code.

Reply:
@rlubos I asked @d3zd3z to take a look and asked whether the cherry-picking was ok.

Do you know @rlubos when Zephyr and the nRF Connect SDK will include MBed TLS 3.1.0?  He was questioning whether we should just apply the fix to 3.1.0 since Zephyr is about to upgrade to it.  Of course, there is then a lag from that until the nRF Connect SDK gets it.

Reply:
According to the information in https://github.com/zephyrproject-rtos/mbedtls/pull/35 we're waiting for mbed TLS 3.1 compatible TFM release, I don't know the details though.

As for the NCS it's a bit different story, as we generally don't rely on mbed TLS from upstream, but rather have our own fork fro nrf_security needs. You'd need to check with our local maintainers from more information.

Reply:
I've heard some good news from the sdk-mbedtls maintainers.  But how should I proceed with this current PR, regarding the need for a compatible Mbed TLS?  Should I move some or all of this PR to sdk-zephyr and mark it noup?

Reply:
We an keep this PR as DNM and open a `sdk-zephyr` counterpart as `fromlist`.

Reply:
OK.

Reply:
@plskeggs I think that would be ideal to revive and finalize this PR since we have the updated mbedTLS now.

Reply:
this PR is up-to-date now @rlubos